packages/multicast_dns/lib/src/resource_record.dart

```diff
@@ -0,0 +1,393 @@
+// Copyright 2018 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:io';
+import 'dart:typed_data';
+
+import 'package:meta/meta.dart';
+import 'package:multicast_dns/src/constants.dart';
+import 'package:multicast_dns/src/packet.dart';
+
+// TODO(dnfield): Probably should go with a real hashing function here
+// when https://github.com/dart-lang/sdk/issues/11617 is figured out.
+const int _seedHashPrime = 2166136261;
+const int _multipleHashPrime = 16777619;
+
+int _combineHash(int current, int hash) =>
+    (current & _multipleHashPrime) ^ hash;
+
+int _hashValues(List<int> values) {
+  assert(values != null);
+  assert(values.isNotEmpty);
+
+  return values.fold(
+    _seedHashPrime,
+    (int current, int next) => _combineHash(current, next),
+  );
+}
+
+/// Enumeration of support resource record types.
+class ResourceRecordType {
+  // This class is intended to be used as a namespace, and should not be
+  // extended directly.
+  factory ResourceRecordType._() => null;
+
+  /// An IPv4 Address record (1).
+  static const int a = 1;
+
+  /// An IPv6 Address record (28).
+  static const int aaaa = 28;
+
+  /// An IP Address reverse map record (12).
+  static const int ptr = 12;
+
+  /// An available service record (33).
+  static const int srv = 33;
+
+  /// A text record (16).
+  static const int txt = 16;
+
+  // TODO(dnfield): Support ANY in some meaningful way.  Might be server only.
+  // /// A query for all records of all types known to the name server.
+  // static const int any = 255;
+
+  /// Asserts that a given int is a valid ResourceRecordtype.
+  static bool debugAssertValid(int resourceRecordType) {
+    return resourceRecordType == a ||
+        resourceRecordType == aaaa ||
+        resourceRecordType == ptr ||
+        resourceRecordType == srv ||
+        resourceRecordType == txt;
+  }
+
+  /// Prints a debug-friendly version of the resource record type value.
+  static String toDebugString(int resourceRecordType) {
+    switch (resourceRecordType) {
+      case a:
+        return 'A (IPv4 Address)';
+      case aaaa:
+        return 'AAAA (IPv6 Address)';
+      case ptr:
+        return 'PTR (Domain Name Pointer)';
+      case srv:
+        return 'SRV (Service record)';
+      case txt:
+        return 'TXT (Text)';
+      default:
```

Comment:
If you remove the default here, then analysis will generate a warning if not all cases are handled (and fail CI).
You can remove the default, and then return "Unknown" after the switch.